richtext-ui/src/main/java/com/zachklipp/richtext/ui/Heading.kt

```diff
@@ -92,7 +94,19 @@ internal val DefaultHeadingStyle: HeadingStyle = { level, textStyle ->
 
   val richTextStyle = currentRichTextStyle.resolveDefaults()
   val headingStyleFunction = richTextStyle.headingStyle!!
-  val currentTextStyle = resolveDefaults(AmbientTextStyle.current, LayoutDirectionAmbient.current)
+
+  // According to [Text] composable's documentation:
+  //   "Additionally, for [color], if [color] is not set, and [style] does not have a color, then
+  //   [AmbientContentColor] will be used - this allows this [Text] or element containing this [Text]
+  //   to adapt to different background colors and still maintain contrast and accessibility."
+  // However, [resolveDefaults] uses a static default color which is [Color.Black].
+  // To fix this issue, we are specifying the text color according to [Text] documentation
+  // before calling [resolveDefaults].
+  val incomingStyle = AmbientTextStyle.current.let {
+    it.copy(color = it.color.useOrElse { AmbientContentColor.current })
```

Comment:
Woah, TIL about useOrElse. I think I'm OK with merging this for now to get the release out, but this is always creating an extra allocation of the text style even if it's not required which is￼ unfortunate.￼